src/System.Security.Cryptography.Pkcs/src/Internal/Cryptography/Pal/AnyOS/ManagedPal.Decrypt.cs

```diff
@@ -82,7 +82,10 @@ private sealed class ManagedDecryptorPal : DecryptorPal
                     }
                     finally
                     {
-                        Array.Clear(cek, 0, cek.Length);
+                        if (cek != null)
+                        {
+                            Array.Clear(cek, 0, cek.Length);
+                        }
```

Comment:
I'm a little confused by this code (the existing code here, not the null check).

We're pinning cek, but it already existed as an unpinned managed array since the earlier `cek = ktri.DecryptCek(cert, out exception)` call, and could have been moved any number of times before we pin it.

Is the pinning just a defense-in-depth to try to minimize its being moved around rather than actually stopping there from being shadows?

Reply:
Yep, it's a best-sane-effort.

Reply:
If it's important, why not go all the way and pin the array (presumably with a GCHandle even though it's less efficient) prior to copying the data into it in DecryptCek?